core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/verticalscale/CoreVerticalScaleActions.java

```diff
@@ -91,8 +105,46 @@ protected void doExecute(ClusterViewContext ctx, CoreVerticalScalingTriggerEvent
                         .withWorkspaceId(stack.getWorkspace().getId())
                         .withAccountId(Crn.safeFromString(stack.getResourceCrn()).getAccountId())
                         .build();
+                CoreVerticalScalePreparationRequest request = new CoreVerticalScalePreparationRequest(
+                        cloudContext,
+                        cloudCredential,
+                        null,
+                        stack,
+                        instanceGroup,
+                        cloudResources,
+                        stackVerticalScaleV4Request);
+                sendEvent(ctx, request);
+            }
+        };
+    }
+
+    @Bean(name = "STACK_VERTICALSCALE_STATE")
+    public Action<?, ?> stackVerticalScale() {
+        return new AbstractClusterAction<>(CoreVerticalScalePreparationResult.class) {
+            @Override
+            protected void doExecute(ClusterViewContext ctx, CoreVerticalScalePreparationResult payload, Map<Object, Object> variables)
+                    throws Exception {
+                StackVerticalScaleV4Request stackVerticalScaleV4Request = payload.getStackVerticalScaleV4Request();
+                StackDto stack = (StackDto) variables.get(STACK_DTO);
```

Comment:
when a flow is active on a stack - no other operations can be performed on it right? and also I was trying to avoid hitting the DB repeatedly for stack information so that is why I was passing the stack along that way. I am not sure how it will become stale as I am not directly changing anything on the stack during flow except in the final step

Reply:
Like I said, since stack is not being modified, I am avoiding multiple DB calls.

Reply:
Leaving this open until another Wolf confirms this shortcut is acceptable. :smile:

Reply:
@doktoric please let us know if this is acceptable. since the stack is not being modified in the middle of flow, and no other operations can be performed on the stack, when a flow is in progress, I think it is OK to put it in the variables map and pass it around.

Reply:
Lajos is right, you should not share Stack and other entities between actions. Flow is not running in a transaction where the state ensured. If you check `AbstractClusterAction#createFlowContext` it would start with fetching the stack from DB.